src/jaeger_passage_reporter.erl

```diff
@@ -123,8 +97,15 @@ start(ReporterId, Options) ->
     Args = [ReporterId, Options],
     is_atom(ReporterId) orelse error(badarg, Args),
     is_list(Options) orelse error(badarg, Args),
-
-    case jaeger_passage_reporter_sup:start_child(ReporterId, Options) of
+    proplists:is_defined(protocol, Options)
+        orelse error(badarg, Args),
+    ReporterModule = case proplists:get_value(protocol, Options) of
```

Comment:
How about specifying the third argument to `proplists:get_value()` and allowing users to omit to give `protocol` parameter explicitly if they want to use the default.

```suggestion
    ReporterModule = case proplists:get_value(protocol, Options, udp) of
```

IMO, if we would like to require users to specify the parameter value explictly, the parameter should be a positional argument instead of a keyword one (i.e., proplists).

Reply:
> IMO, if we would like to require users to specify the parameter value explictly, the parameter should be a positional argument instead of a keyword one (i.e., proplists).

Do you mean changing arity of jaeger_passage_reporter:start function?
```
start(ReporterId, Protocol) ->
    start(ReporterId, Protocol, []).

start(ReporterId, Protocol, Options) ->
    ...
```
I agree this is cleaner change. However this would be backward incompatible change. Please confirm that you want me to do it.

Reply:
Sorry for my ambiguous comment, but didn't intend to suggest the change. What I wanted to say is that **`protocol` parameter should be optional**. Due to the usage of `proplists:get_value/2`, the current implementation requires users to specify `protocol` parameter explicitly. If users omit the parameter, a badarg error will be raised as follows:
```erlang
$ git clone git://github.com/iilyak/jaeger_passage -b add-http-reporter
$ cd jaeger_passage
$ rebar3 shell
1> jaeger_passage_reporter:start(foo, []).
** exception error: bad argument
     in function  jaeger_passage_reporter:start/2
        called as jaeger_passage_reporter:start(foo,[])
```